astropy/coordinates/sky_coordinate.py

```diff
@@ -72,9 +72,11 @@ def _represent_as_dict(self):
         if 'distance' in attrs and np.all(obj.distance == 1.0):
             attrs.remove('distance')
 
-        self._represent_as_dict_attrs = attrs
```

Comment:
One is no longer allowed to change this - but it is also not necessary any more, so a change for the better I think.